File: src/solutions/01/ts/01.ts
import path from "path";
import { readInputFile } from "../../../lib/input";

async function part01() {
  const input = await readInputFile(path.join(__dirname, "../data.txt"));
  const data = input.split("\n").map(Number);

  for (let x = 0; x < data.length; x++) {
    for (let y = x + 1; y < data.length; y++) {
      if (data[x] + data[y] === 2020) {
        console.log(`Part 1: ${data[x] * data[y]}`);
      }
    }
  }
}

async function part02() {
  const input = await readInputFile(path.join(__dirname, "../data.txt"));
  const data = input.split("\n").map(Number);

  for (let x = 0; x < data.length; x++) {
    for (let y = x + 1; y < data.length; y++) {
      for (let z = y + 1; z < data.length; z++) {
        if (data[x] + data[y] + data[z] === 2020) {
          console.log(`Part 2: ${data[x] * data[y] * data[z]}`);
        }
      }
    }
  }
}

part01();
part02();

File: src/solutions/01/rust/01.rs
use crate::lib::input;

pub fn day01() {
    let input_data = input::load_input("src/solutions/01/data.txt");
    let data = input_data
        .lines()
        .map(|l| l.parse().unwrap())
        .collect::<Vec<u32>>();

    println!("Day 1:");
    for x in 0..data.len() {
        for y in x + 1..data.len() {
            if data[x] + data[y] == 2020 {
                println!("Part 1: {}", data[x] * data[y]);
            }

            for z in y + 1..data.len() {
                if data[x] + data[y] + data[z] == 2020 {
                    println!("Part 2: {}", data[x] * data[y] * data[z]);
                }
            }
        }
    }
}
